l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -0,0 +1,249 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+
+import "./IBridgehub.sol";
+import "../bridge/interfaces/IL1Bridge.sol";
+import "../state-transition/IStateTransitionManager.sol";
+import "../common/ReentrancyGuard.sol";
+import "../state-transition/chain-interfaces/IZkSyncStateTransition.sol";
+
+contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2Step {
+    address public constant ETH_TOKEN_ADDRESS = address(1);
+
+    /// new fields
+    /// @notice we store registered stateTransitionManagers
+    mapping(address => bool) public stateTransitionManagerIsRegistered;
+    /// @notice we store registered tokens (for arbitrary base token)
+    mapping(address => bool) public tokenIsRegistered;
+    /// @notice we store registered bridges
+    mapping(address => bool) public tokenBridgeIsRegistered;
+
+    /// @notice chainID => stateTransitionManager contract address
+    mapping(uint256 => address) public stateTransitionManager;
+    /// @notice chainID => base token address
+    mapping(uint256 => address) public baseToken;
+    /// @notice chainID => bridge holding the base token
+    /// @notice a bridge can have multiple tokens
+    mapping(uint256 => address) public baseTokenBridge;
```

Comment:
All three mappings are indexed by chain ID and store address. The address is 20 bytes long, while 1 storage slot is 32 bytes long. If we use 1 mapping from chain ID to struct with 3 addreses we can save 1 storage slot.